### ISSUE TITLE
why we need the first indexs2.shape[0] in gt_bo_masks[:indexs2.shape[0]]?

### ISSUE THREAD

**Issue by hellodhd** [lkeab/BCNet#48]
Thanks for your attention, codes as follows.

new_gt_bo_masks1 = gt_bo_masks[indexs2,:,:].squeeze()
new_gt_bo_masks2 = gt_bo_masks[:indexs2.shape[0]]
if new_gt_bo_masks1.shape != new_gt_bo_masks2.shape:
        new_gt_bo_masks1 = new_gt_bo_masks1.unsqueeze(0)
new_gt_bo_masks = torch.cat((new_gt_bo_masks1, new_gt_bo_masks2),0)

**Comment by lkeab** [comment]
Thanks for the question.

This [code section](https://github.com/lkeab/BCNet/blob/main/detectron2/modeling/roi_heads/mask_head.py#L129) is to balance the number of occluded objects vs. non-occluded object during mask loss training. It is beneficial for the first GCN layer for predicting the occluders by making occluded objects occupying over 50%.
